mypy_django_plugin/transformers/managers.py

```diff
@@ -71,25 +72,37 @@ def get_method_type_from_dynamic_manager(
     queryset_info = helpers.lookup_fully_qualified_typeinfo(api, queryset_fullname)
     assert queryset_info is not None
 
-    def get_funcdef_type(definition: Union[FuncBase, Decorator, None]) -> Optional[ProperType]:
-        # TODO: Handle @overload?
-        if isinstance(definition, FuncBase) and not isinstance(definition, OverloadedFuncDef):
-            return definition.type
-        elif isinstance(definition, Decorator):
-            return definition.func.type
-        return None
+    is_fallback_queryset = queryset_info.metadata.get("django", {}).get("any_fallback_queryset", False)
 
-    method_type = get_funcdef_type(queryset_info.get_method(method_name))
-    if method_type is None:
-        return None
+    method_type = _get_funcdef_type(queryset_info.get_method(method_name))
+    if not isinstance(method_type, FunctionLike):
+        return method_type
+
+    items = []
+    for item in method_type.items:
+        items.append(
+            _process_dynamic_method(
+                method_name,
+                item,
+                queryset_info=queryset_info,
+                manager_instance=manager_instance,
+                is_fallback_queryset=is_fallback_queryset,
+            )
+        )
+    return Overloaded(items) if len(items) > 1 else items[0]
 
-    assert isinstance(method_type, CallableType)
 
+def _process_dynamic_method(
+    method_name: str,
+    method_type: CallableType,
+    *,
```

Comment:
Oh, nice, I just assumed that we still supported too old versions that we couldn't force named arguments.

I'm quite certain I've seen a couple of boolean traps in the plugin code

Reply:
I will hunt them down. Might even try to lint them away with ruff: https://beta.ruff.rs/docs/rules/#flake8-boolean-trap-fbt

Reply:
Some many rules stolen from the one true linter 😆